File: Python/programmers/etc/level3/입국심사.py
def solution(n, times):
    left, right = 0, max(times)*n
    while left+1 < right:
        mid = (left+right)//2
        tmp = 0
        for t in times:
            tmp += mid//t
        if tmp >= n:
            right = mid
        else:
            left = mid
    return right


n = 6
times = [7, 10]
print(solution(n, times)) # 28

File: Java/programmers/etc/level3/입국심사.java
package CodeTest.Java.programmers.etc.level3;

public class 입국심사 {
    public long solution(int n, int[] times) {
        int _max = 0;
        for (int time : times) {
            _max = Math.max(_max, time);
        }
        
        long answer = 0;
        long left = 0, right = (long) _max*n;
        
        while (left <= right) {
            long mid = (left+right)/2;
            long tmp = 0;
            for (int time : times) {
                tmp += mid/time;
            }
            if (tmp >= n) {
                right = mid-1;
                answer = mid;
            } else {
                left = mid+1;
            }
        }
        
        return answer;
    }
}
